scipy/stats/_distn_infrastructure.py

```diff
@@ -3245,6 +3198,103 @@ def _iter_chunked(x0, x1, chunksize=4, inc=1):
         yield supp
 
 
+class rv_sample(rv_discrete):
+    """A 'sample' discrete distribution defined by the support and values.
+
+       The ctor ignores most of the arguments, only needs the `values` argument.
+    """
+    def __init__(self, a=0, b=inf, name=None, badvalue=None,
+                 moment_tol=1e-8, values=None, inc=1, longname=None,
+                 shapes=None, extradoc=None, seed=None):
+
+        super(rv_discrete, self).__init__(seed)
+
+        if values is None:
+            raise ValueError("rv_sample.__init__(..., values=None,...)")
+
+        # cf generic freeze
+        self._ctor_param = dict(
+            a=a, b=b, name=name, badvalue=badvalue,
+            moment_tol=moment_tol, values=values, inc=inc,
+            longname=longname, shapes=shapes, extradoc=extradoc, seed=seed)
+
+        if badvalue is None:
+            badvalue = nan
+        self.badvalue = badvalue
+        self.moment_tol = moment_tol
+        self.inc = inc
+        self.shapes = shapes
+        self.vecentropy = self._entropy
+
+        self.xk, self.pk = values
+        self.return_integers = 0
+        indx = np.argsort(np.ravel(self.xk))
+        self.xk = np.take(np.ravel(self.xk), indx, 0)
+        self.pk = np.take(np.ravel(self.pk), indx, 0)
+        self.a = self.xk[0]
+        self.b = self.xk[-1]
+        self.qvals = np.cumsum(self.pk, axis=0)
+
+        self.shapes = ' '   # bypass inspection
+        self._construct_argparser(meths_to_inspect=[self._pmf],
+                                  locscale_in='loc=0',
+                                  # scale=1 for discrete RVs
+                                  locscale_out='loc, 1')
+
+        self._construct_docstrings(name, longname, extradoc)
+
+    def _pmf(self, x):
+        return np.select([x == k for k in self.xk],
+                         [np.broadcast_arrays(p, x)[0] for p in self.pk], 0)
+
+    def _cdf(self, x):
+        xx, xxk = np.broadcast_arrays(x[:, None], self.xk)
+        indx = np.argmax(xxk > xx, axis=-1) - 1
+        return self.qvals[indx]
+
+    def _ppf(self, q):
+        qq, sqq = np.broadcast_arrays(q[:, None], self.qvals)
+        indx = argmax(sqq >= qq, axis=-1)
+        return self.xk[indx]
+
+    def _rvs(self):
+        # Need to define it explicitly, otherwise .rvs() with size=None
+        # fails due to explicit broadcasting 
```

Comment:
Alternatively, could you add ``q = np.atleast_1d(q)`` in ``_ppf`` and avoid having to add this whole method?